lib/dockerregistry/inventory.go

```diff
@@ -1952,6 +1952,38 @@ func MKPopulateRequestsForPromotionEdges(
 	}
 }
 
+// CheckPromotion is where PreChecks to be run are defined.
```

Comment:
I think you can delete `CheckPromotion` and just use sc.RunChecks directly at line 350.